tests/fixture/tmpnet/process_runtime.go

```diff
@@ -235,6 +236,26 @@ func (p *ProcessRuntime) WaitForStopped(ctx context.Context) error {
 	}
 }
 
+// Restarts the node
+func (p *ProcessRuntime) Restart(ctx context.Context) error {
+	if p.getRuntimeConfig().ReuseDynamicPorts {
+		// Attempt to save the API port currently being used so the
+		// restarted node can reuse it. This may result in the node
+		// failing to start if the operating system allocates the port
+		// to a different process between node stop and start.
+		if err := p.saveAPIPort(); err != nil {
+			return err
+		}
+	}
```

Comment:
out of scope / no action required: should we have a fallback to use another port (aka `0`) if the start fails due to the port being assigned to another process? Although this is quite unlikely to happen still, hence feel free to ignore this comment!